internal/v5/api.go

```diff
@@ -1466,10 +1467,54 @@ func (h *ReqHandler) servePromulgate(id *router.ResolvedURL, w http.ResponseWrit
 	return nil
 }
 
+// GET id/allperms
+// https://github.com/juju/charmstore/blob/v5/docs/API.md#get-id-allperms
+func (h *ReqHandler) serveAllPerms(id *charm.URL, w http.ResponseWriter, req *http.Request) error {
+	if req.Method != "GET" {
+		return errgo.WithCausef(nil, params.ErrMethodNotAllowed, "%s not allowed", req.Method)
+	}
+	if id.Revision != -1 {
```

Comment:
Any particular reason why not? especially for the promulgated one.

Reply:
17:59:22] <rogpeppe1> mhilton: for the promulgated one it's ambiguous which base entity you want
[17:59:38] <rogpeppe1> mhilton: as a promulgated charm can have several base entities
[17:59:56] <rogpeppe1> mhilton: so i thought it best to require explicitness
[17:59:57] <mhilton> rogpeppe1, surely it's the one with promulgated set to True
[18:00:38] <mhilton> rogpeppe1, if you don't need the functionality I'm sure it can be added later anyway. I was mainly curious
[18:00:48] <rogpeppe1> mhilton: yeah, i guess so
[18:00:54] <rogpeppe1> mhilton: i don't need the functionality :)
[18:01:17] <rogpeppe1> mhilton: and it would be a bit of hassle to do, i think